roms_tools/setup/cdr_forcing.py

```diff
@@ -873,11 +693,25 @@ def to_yaml(self, filepath: Union[str, Path]) -> None:
             The path to the YAML file where the parameters will be saved.
         """
 
-        _to_yaml(self, filepath)
+        # Serialize object into dictionary
+        yaml_data = _to_dict(self)
```

Comment:
ditto on naming as `yaml_data` when it's a dictionary. you're naming a variable based on who is going to use it, not what it contains.